sdk/src/worker.ts

```diff
@@ -0,0 +1,145 @@
+import { initializeWasm, initThreadPool, ProgramManager, PrivateKey, verifyFunctionExecution } from "./browser";
+import { AleoKeyProvider, AleoKeyProviderParams} from "./function-key-provider";
+import { expose } from "comlink";
+
+await initializeWasm();
+await initThreadPool(10);
+
+const defaultHost = "https://vm.aleo.org/api";
+const keyProvider = new AleoKeyProvider();
+const programManager = new ProgramManager(
+    defaultHost,
+    keyProvider,
+    undefined
+);
+
+keyProvider.useCache(true);
+
+// self.postMessage({
+//     type: "ALEO_WORKER_READY",
+// });
+
+export interface WorkerAPI {
+    additionalFunction1: () => Promise<string>;
+    additionalFunction2: (data: any) => Promise<any>;
+}
+
+let lastLocalProgram: string = "";
+
+export interface WorkerAPI {
+    executeOffline: (
+        localProgram: string,
+        aleoFunction: string,
+        inputs: string[],
+        privateKey: string
+    ) => Promise<{ outputs: any; execution: string } | string>;
+
+    getPrivateKey: () => Promise<PrivateKey>;
+}
+async function executeOffline(
+    localProgram: string,
+    aleoFunction: string,
+    inputs: string[],
+    privateKey: string
```

Comment:
```suggestion
    privateKey: string
    proveExecution: boolean = false
```

We should have the prove execution here, default false and propagate that below